udata/core/dataset/models.py

```diff
@@ -587,7 +603,7 @@ def last_update(self):
         if self.resources:
             dates = []
             for res in self.resources:
-                dates.append(res.modified)
+                dates.append(res.last_modified)
                 if res.harvest and res.harvest.modified_at:
                     dates.append(to_naive_datetime(res.harvest.modified_at))
```

Comment:
I think we can refactor this to use `res.last_modified` that takes the max for us :)